vault/login_mfa.go

```diff
@@ -1997,7 +1999,7 @@ func (c *Core) validatePingID(ctx context.Context, mConfig *mfa.Config, username
 	return nil
 }
 
-func (c *Core) validateTOTP(ctx context.Context, creds []string, entityMethodSecret *mfa.Secret, configID, entityID string) error {
+func (c *Core) validateTOTP(ctx context.Context, creds []string, entityMethodSecret *mfa.Secret, configID, entityID string, usedCodes *cache.Cache, maximumValidationAttempts int64) error {
```

Comment:
I am wondering if this feature is also needed for the PolicyMFA? Although it may break backwards compatibility for PolicyMFA, since the root problem is a security issue, maybe we would want to have it in PolicyMFA as well? 
For now, this would break PolicyMFA call to validateTOTP and I would need to push a PR on the ENT side. If this improvement is not needed for PolicyMFA, I will revert the change to the validateTOTP function.

Reply:
Yeah, I thought it might be an exception since this would improve security of the existing ENT Policy MFA as well. So, not sure to keep it or to get rid of it. @mickael-hc any thoughts?